vault/rollback.go

```diff
@@ -244,18 +264,27 @@ func (m *RollbackManager) finishRollback(rs *rollbackState, err error, fullPath
 		m.inflightLock.Lock()
 		defer m.inflightLock.Unlock()
 	}
-	delete(m.inflight, fullPath)
+	if _, ok := m.inflight[fullPath]; ok {
+		delete(m.inflight, fullPath)
+	}
 }
 
 // attemptRollback invokes a RollbackOperation for the given path
 func (m *RollbackManager) attemptRollback(ctx context.Context, fullPath string, rs *rollbackState, grabStatelock bool) (err error) {
+	close(rs.isRunning)
```

Comment:
Missing a lot of context here, but I found it interesting that we close this channel before the rollback is finished.

In `(*RollbackManager).Rollback`, we check to see if there's a `isRunning` and if not, proceed on to close the channel. Could there be a race/panic here where we double-close?

Reply:
isRunning is only ever closed in one place, here in the attemptRollback method.

isCancelled is also only ever closed in one place, in the Rollback method. 

We check to see if these channels are closed in other places by doing 
```
select {
  case <-rs.isRunning:
  ...
}
```

but that action won't close the channel, so there's no risk of double close. 

I'm happy to jump on a call if you'd like to talk through the concurrency.